### ISSUE TITLE
System Color problem

### ISSUE THREAD

**Issue by phoenixfenix** [bgrabitmap/bgracontrols#26]
Hello,
using the debugger (FPC 3.0.4 and Lazarus 1.8.4 (Win10) 32bit) with the controls enabled, I detected a "RunError 201" error on ColorToBGRA () when I click to change the value of a BCTrackbarUpdown control.
I think I found the cause:

in TCustomBCTrackbarUpdown.RenderOnBitmap ()

```
ABitmap.FillRect(x,y+1,x+s.cx,y+s.cy,clHighlight,dmSet);
ABitmap.TextOut(x,y,inSel,clHighlightText);
```

these lines should be:

```
ABitmap.FillRect(x,y+1,x+s.cx,y+s.cy,ColorToRGB(clHighlight),dmSet);
ABitmap.TextOut(x,y,inSel,ColorToRGB(clHighlightText));
```

 Thanks

**Comment by lainz** [comment]
We have automatic conversion between TBGRAPixel and TColor, that is not working fine?

**Comment by phoenixfenix** [comment]
Without the change in debugger mode it stops on:
function ColorToBGRA (color: TColor): TBGRAPixel;
in particular, it seems that a negative value is assigned to a cardinal.

ColorToBGRA call:
procedure RedGreenBlue(rgb: TColor; out Red, Green, Blue: Byte); inline;// does not work on system color 

rgb: TColor = TGraphicsColor = -$7FFFFFFF-1..$7FFFFFFF;

```
procedure RedGreenBlue(rgb: TColor; out Red, Green, Blue: Byte);
begin
  (*if rgb < 0 then //if enabled it avoids debugger error (without the modification in the control code)
   Exit;*)
  Graphics.RedGreenBlue(rgb, Red, Green, Blue);
end; 
```

Graphics.RedGreenBlue() requires parameter 
rgb: TColorRef = COLORREF = Cardinal (LCLType unit)

**Comment by lainz** [comment]
Hi ok. It should be reported in bgrabitmap instead.

**Comment by lainz** [comment]
Applied your fix in commit 32bec09fba97c59b8550fac8f24ce5f53d09e0c1

**Comment by phoenixfenix** [comment]
Thanks!
